apache/zeppelin
# [ZEPPELIN-2698] Apply KerberosInterpreter to JDBCInterpreter

### What is this PR for?
This is to apply new KerberosInterpreter mechanism to JDBCInterpreter for relogin from keytab, instead of on fail relogin.



### What type of PR is it?
[Refactoring]

### What is the Jira issue?
* [ZEPPELIN-2698](https://issues.apache.org/jira/browse/ZEPPELIN-2698)

### How should this be tested?
In JDBC interpreter setting add following properties
- zeppelin.jdbc.auth.type = KERBEROS
- zeppelin.jdbc.principal = principal value
- zeppelin.jdbc.keytab.location = keytab location

Now try and run any of hive's query (say `show tables`) it should return with valid results.
Again, wait for this kerberos ticket to expire (usually its 24hrs) then try the above again, and it should work.


### Questions:
* Does the licenses files need update? N/A
* Is there breaking changes for older versions? N/A
* Does this needs documentation? N/A
